osu.Game/Screens/Play/Player.cs

```diff
@@ -820,6 +823,40 @@ protected override bool OnScroll(ScrollEvent e)
             return mouseWheelDisabled.Value && !e.AltPressed;
         }
 
+        #region Gameplay leaderboard
+
+        protected readonly Bindable<bool> LeaderboardExpandedState = new BindableBool();
+
+        private void loadLeaderboard()
+        {
+            HUDOverlay.HoldingForHUD.BindValueChanged(_ => updateLeaderboardExpandedState());
+            LocalUserPlaying.BindValueChanged(_ => updateLeaderboardExpandedState(), true);
+
+            LoadComponentAsync(CreateGameplayLeaderboard(), leaderboard =>
+            {
+                if (!LoadedBeatmapSuccessfully)
+                    return;
+
+                leaderboard.Expanded.BindTo(LeaderboardExpandedState);
+
+                AddLeaderboardToHUD(leaderboard);
+            });
+        }
+
+        public readonly BindableList<ScoreInfo> LeaderboardScores = new BindableList<ScoreInfo>();
+
+        protected virtual GameplayLeaderboard CreateGameplayLeaderboard() => new SoloGameplayLeaderboard(Score.ScoreInfo.User)
```

Comment:
In addition to my comment above about `LeaderboardScores` bindable, I would make this abstract and override in `SoloPlayer`/`ReplayPlayer`. `Player` shouldn't create a "solo" component.

Reply:
yes well. i was avoiding having to do this in every implementation except multiplayer. i guess it's fine to do so though.